openedx/openedx-platform
# Campus.il - SE-1152 Hide CCX courses from catalog

This PR hides all ccx courses from the course catalog. Previously the catalog visibility was inherited from the base course at the time of ccx creation, resulting in the possibility of ccx courses to be unexpectedly visible in the catalog.

**JIRA tickets**: [OSPR-3625](https://openedx.atlassian.net/browse/OSPR-3625)

**Dependencies**: None

**Sandbox URL**:

-    LMS: https://pr20746.sandbox.opencraft.hosting/
-    Studio: https://studio.pr20746.sandbox.opencraft.hosting/


**Merge deadline**: None

**Testing instructions**:

1. Enable CCX courses (`CUSTOM_COURSES_EDX: true`)
2. go through the process of [creating a custom course](https://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/set_up_course/studio_add_course_information/custom_courses.html).
3. log in as a non-staff user and view the course catalog page. the ccx course overview should not be seen.
4. also test with `COURSE_CATALOG_VISIBILITY_PERMISSION` set to `see_in_catalog`

**Author notes and concerns**:

- The default catalog/about visibility permissions are `see_exists`, which renders the `catalog_visibility` advanced course setting useless.

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
    CUSTOM_COURSES_EDX: true
```
